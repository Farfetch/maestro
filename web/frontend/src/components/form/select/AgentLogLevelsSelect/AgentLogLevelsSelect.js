import { Button, Select } from "antd";
import React, { useState } from "react";

import { agentLogLevel } from "../../../../lib/api/models";

const AgentLogLevelsSelect = ({ onChange = () => {} }) => {
  const [selectedLevels, setSelectedLevels] = useState([]);

  const handleSelectAll = () => {
    const allLevels = Object.values(agentLogLevel);
    setSelectedLevels(allLevels);
    onChange(allLevels);
  };

  const handleUnselectAll = () => {
    setSelectedLevels([]);
    onChange([]);
  };

  const handleChange = (selected) => {
    setSelectedLevels(selected);
    onChange(selected);
  };

  return (
    <div style={{ display: "flex", alignItems: "center" }}>
      <Select
        mode={"multiple"}
        allowClear
        placeholder="Select Log level"
        onChange={handleChange}
        style={{ width: "100%" }}
        defaultValue={agentLogLevel.DEBUG}
        value={selectedLevels}
      >
        {Object.values(agentLogLevel).map((logLevel) => (
          <Select.Option key={logLevel}>{logLevel}</Select.Option>
        ))}
      </Select>
      <Button
        type="primary"
        onClick={handleSelectAll}
        style={{ marginLeft: "10px" }}
      >
        Select All
      </Button>
      <Button
        type="danger"
        onClick={handleUnselectAll}
        style={{ marginLeft: "10px" }}
      >
        Unselect All
      </Button>
    </div>
  );
};

export default AgentLogLevelsSelect;
