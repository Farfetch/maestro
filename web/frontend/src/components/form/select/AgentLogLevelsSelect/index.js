import AgentLogLevelsSelect from "./AgentLogLevelsSelect";

export default AgentLogLevelsSelect;
