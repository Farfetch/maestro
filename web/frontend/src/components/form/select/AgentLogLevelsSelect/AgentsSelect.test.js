import { render } from "@testing-library/react";
import React from "react";

import AgentLogLevelsSelect from "./AgentLogLevelsSelect";

describe("components/form/select/AgentLogLevelsSelect", () => {
  ["single", "multiple"].map((mode) =>
    test(`should render AgentLogLevelsSelect component with mode=${mode}`, async () => {
      const rendered = render(<AgentLogLevelsSelect mode={mode} />);

      const badgeComponent = rendered.container;
      expect(badgeComponent.outerHTML).toMatchSnapshot();
    })
  );
});
