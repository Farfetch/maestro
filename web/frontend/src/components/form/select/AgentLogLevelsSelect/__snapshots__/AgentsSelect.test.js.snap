// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/form/select/AgentLogLevelsSelect should render AgentLogLevelsSelect component with mode=multiple 1`] = `"<div><div style=\\"display: flex; align-items: center;\\"><div class=\\"ant-select ant-select-multiple ant-select-allow-clear ant-select-show-search\\" style=\\"width: 100%;\\"><div class=\\"ant-select-selector\\"><div class=\\"ant-select-selection-overflow\\"><div class=\\"ant-select-selection-overflow-item ant-select-selection-overflow-item-suffix\\" style=\\"opacity: 1;\\"><div class=\\"ant-select-selection-search\\" style=\\"width: 0px;\\"><input type=\\"search\\" autocomplete=\\"off\\" class=\\"ant-select-selection-search-input\\" role=\\"combobox\\" aria-haspopup=\\"listbox\\" aria-owns=\\"rc_select_TEST_OR_SSR_list\\" aria-autocomplete=\\"list\\" aria-controls=\\"rc_select_TEST_OR_SSR_list\\" aria-activedescendant=\\"rc_select_TEST_OR_SSR_list_0\\" readonly=\\"\\" unselectable=\\"on\\" style=\\"opacity: 0;\\" value=\\"\\" id=\\"rc_select_TEST_OR_SSR\\"><span class=\\"ant-select-selection-search-mirror\\" aria-hidden=\\"true\\">&nbsp;</span></div></div></div><span class=\\"ant-select-selection-placeholder\\">Select Log level</span></div></div><button style=\\"margin-left: 10px;\\" type=\\"button\\" class=\\"ant-btn ant-btn-primary\\"><span>Select All</span></button><button style=\\"margin-left: 10px;\\" type=\\"button\\" class=\\"ant-btn ant-btn-danger\\"><span>Unselect All</span></button></div></div>"`;

exports[`components/form/select/AgentLogLevelsSelect should render AgentLogLevelsSelect component with mode=single 1`] = `"<div><div style=\\"display: flex; align-items: center;\\"><div class=\\"ant-select ant-select-multiple ant-select-allow-clear ant-select-show-search\\" style=\\"width: 100%;\\"><div class=\\"ant-select-selector\\"><div class=\\"ant-select-selection-overflow\\"><div class=\\"ant-select-selection-overflow-item ant-select-selection-overflow-item-suffix\\" style=\\"opacity: 1;\\"><div class=\\"ant-select-selection-search\\" style=\\"width: 0px;\\"><input type=\\"search\\" autocomplete=\\"off\\" class=\\"ant-select-selection-search-input\\" role=\\"combobox\\" aria-haspopup=\\"listbox\\" aria-owns=\\"rc_select_TEST_OR_SSR_list\\" aria-autocomplete=\\"list\\" aria-controls=\\"rc_select_TEST_OR_SSR_list\\" aria-activedescendant=\\"rc_select_TEST_OR_SSR_list_0\\" readonly=\\"\\" unselectable=\\"on\\" style=\\"opacity: 0;\\" value=\\"\\" id=\\"rc_select_TEST_OR_SSR\\"><span class=\\"ant-select-selection-search-mirror\\" aria-hidden=\\"true\\">&nbsp;</span></div></div></div><span class=\\"ant-select-selection-placeholder\\">Select Log level</span></div></div><button style=\\"margin-left: 10px;\\" type=\\"button\\" class=\\"ant-btn ant-btn-primary\\"><span>Select All</span></button><button style=\\"margin-left: 10px;\\" type=\\"button\\" class=\\"ant-btn ant-btn-danger\\"><span>Unselect All</span></button></div></div>"`;
